src/ChatGptNet/ChatGptClientFactory.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Microsoft.Extensions.Caching.Memory;
+
+namespace ChatGptNet;
+
+internal class ChatGptClientFactory : IChatGptClientFactory
+{
+    private readonly IMemoryCache memoryCache;
+    private readonly ChatGptOptions defaultOptions;
+
+    public ChatGptClientFactory(IMemoryCache memoryCache, ChatGptOptions defaultOptions)
+    {
+        this.memoryCache = memoryCache;
+        this.defaultOptions = defaultOptions;
+    }
+
+    public IChatGptClient CreateClient(Action<ChatGptOptions> setupAction)
```

Comment:
The parameter should be nullable and optional, if I want to create a client with the default options and I don't need to change anything else. Moreover, an overload the accepts a `ServiceProvider` as argument is needed, to cover a scenario like this one: https://github.com/marcominerva/ChatGptNet/blob/2193d2c7b9656d337cbd24b344b8ea3bba995fbb/src/ChatGptNet/ChatGptServiceCollectionExtensions.cs#L72

It is necessary when I need to provide options that have external dependencies (for example, an Api Key that depends on the current user).

Reply:
Implemented the `CreateClient` method with 3 overloads: a parameterless version that uses the default options, plus two overloads that accept an `Action<ChatGptOptions>` or `Action<IServiceProvider, ChatGptOptions>`